### PR TITLE
file_packager: Fix regression in --separate-metadata output

### PR DESCRIPTION
**Pull request by sbc100**
This reverts commit 483cdcd5306780520a2de8856fcd13910e71497e.

This also adds some testing of the `--separate-metadata` file_packager options where we actually run the generated code.  As part of adding this test I also made the code work under node.

Fixes: #22790

### CODE REVIEW

**Review by kripken — APPROVED** [review]
Can we add a test?

**Comment by sbc100** [comment]
I've added a test as a separate commit.  It also fixes `--separate-metadata` so that it runs under node.  OK to land these together or should I land the test after the revert?

File: test/test_other.py [hunk 1]
```diff
@@ -3620,6 +3620,21 @@ def test_file_packager_separate_metadata(self):
...
+    int main() {
+      struct stat buf;
+      assert(stat("data1.txt", &buf) == 0);
+      assert(stat("subdir/data2.txt", &buf) == 0);
+      printf("done\n");
+      return 0;
+    }
+    ''')
+    self.do_runf('src.c', emcc_args=['--pre-js=immutable.js', '-sFORCE_FILESYSTEM'])
+
```
Comment (kripken) [inline test/test_other.py:3637]: I'm not opposed to adding the test + node fix in this PR, that sgtm, but can you add an explanation for the test? It just appends to an existing test and I can't figure out what specifically it is testing, and why it failed before this reversion.

Comment (kripken) [reply test/test_other.py:3637]: I mean that it looks like many other tests we already have coverage for, so I'm missing what is special here.

Comment (sbc100) [reply test/test_other.py:3637]: There were no other node tests that ran with `--separate-metadata`.   It was that options specifically that was broken.

**Comment by kripken** [comment]
Thanks, lgtm

**Pull request merged**